### ISSUE TITLE
trio-websocket hangs on custom streams

### ISSUE THREAD

**Issue by A5rocks** [HyperionGray/trio-websocket#131]
This code doesn't work:
```py
import uuid
import logging
import trio_websocket
import trio.testing
import trio
import typing

CONNECTED_WS_TUPLE = typing.Tuple[trio_websocket.WebSocketConnection, trio_websocket.WebSocketConnection]


async def connected_websockets(nursery: trio.Nursery) -> CONNECTED_WS_TUPLE:
    left, right = trio.testing.memory_stream_pair()

    left = await trio_websocket.wrap_client_stream(nursery, left, '<URL WITHOUT "wss://">', '/')
    right = await trio_websocket.wrap_server_stream(nursery, right)

    return left, right

logging.basicConfig(level='DEBUG')


async def test_correctness(nursery):
    left, right = await connected_websockets(nursery)

    message = str(uuid.uuid4())
    await left.send_message(message)
    assert await right.get_message() == message


async def main():
    async with trio.open_nursery() as nursery:
        nursery.start_soon(test_correctness, nursery)

trio.run(main)
```

As far as I can see, there shouldn't be anything wrong with what I did, but you never know. My problem is that all the program does is this:

```
$ python test_mockws.py
DEBUG:trio-websocket:client-0 sending 157 bytes
```

I would think the only special thing is that I am using `trio.testing.memory_stream_pair()`, but that returns a `StapledStream`, which is a stream type.

TL;DR `trio.testing.memory_stream_pair` and `trio_websocket.wrap_client/server_stream` don't seem to be working together very well. But they should.

**Comment by A5rocks** [comment]
From the little digging I just did, the cause seems to be `await right.get_message()`.

Code reference: https://github.com/HyperionGray/trio-websocket/blob/master/trio_websocket/_impl.py#L821

**Comment by A5rocks** [comment]
Correction: It seems `await left.send_message(message)` never returns. I have no idea what may be causing this behavior. 

Something to consider is that `await self._send(event)` **does** in fact return. That's the last statement in `send_message` though, so like I said, I have no ideas as for what is happening.

**Comment by A5rocks** [comment]
Sorry, it seems I was being dumb.
